src/Downloader.ts

```diff
@@ -184,37 +184,56 @@ class Downloader {
     return `${cachedPart}${path}`;
   }
 
-  public async checkCapabilities(): Promise<void> {
-    // check if RESTBase-powered API available
+  public setBaseUrls(baseUrl: string, baseUrlForMainPage: string = this.mw.veApiUrl.href){
+    this.baseUrl = baseUrl;
+    this.baseUrlForMainPage = baseUrlForMainPage;
+  }
+
+  public async checkRestApiAvailabilty(type: string, restApiBase: string, defaultUrlForMainPage: string){
     try {
-      const restApiMainPageQuery = await this.getJSON<any>(`${this.baseUrl}${encodeURIComponent(this.mw.metaData.mainPage)}`);
-      this.mwCapabilities.restApiAvailable = !!restApiMainPageQuery.lead;
+      const restBaseApiQuery = await this.getJSON<any>(`${restApiBase}${type}${encodeURIComponent(this.mw.metaData.mainPage)}`);
+      this.setBaseUrls(`${restApiBase}${type}`);
+      (type === 'mobile-sections/') ?  this.mwCapabilities.mobileRestApiAvailable = !!restBaseApiQuery.lead : this.mwCapabilities.desktopRestApiAvailable = !!restBaseApiQuery;
     } catch (err) {
-      this.mwCapabilities.restApiAvailable = false;
-      logger.warn(`Failed to get remote Rest API`);
+      (type === 'mobile-sections/') ?  this.mwCapabilities.mobileRestApiAvailable = false : this.mwCapabilities.desktopRestApiAvailable = false;
+      logger.warn(`Failed to get remote ${type} API`);
     }
+  }
+
+  public async checkCapabilities(): Promise<void> {
+    // check if RESTBase-powered API available
+    const restApiBase = `${this.mw.restApiUrl.href}page/`;
+    const defaultUrlForMainPage = this.mw.veApiUrl.href;
+
+    await this.checkRestApiAvailabilty('html/', restApiBase, defaultUrlForMainPage);
+    await this.checkRestApiAvailabilty('mobile-sections/', restApiBase, defaultUrlForMainPage);
 
     if (!this.forceLocalParser) {
       // check if VisualEditor available
       try {
         const parsoidMainPageQuery = await this.getJSON<any>(`${this.mw.veApiUrl.href}${encodeURIComponent(this.mw.metaData.mainPage)}`);
         this.mwCapabilities.veApiAvailable = !!parsoidMainPageQuery.visualeditor.content;
+
+        if(!this.mwCapabilities.desktopRestApiAvailable && !this.mwCapabilities.mobileRestApiAvailable){
+          this.setBaseUrls(defaultUrlForMainPage);
+        }
       } catch (err) {
         this.mwCapabilities.veApiAvailable = false;
         logger.warn(`Failed to get remote Parsoid`);
       }
     }
 
     if (!this.noLocalParserFallback) {
-      if (!this.mwCapabilities.restApiAvailable || !this.mwCapabilities.veApiAvailable) {
-        logger.log(`Using local MCS and ${this.mwCapabilities.veApiAvailable ? 'remote' : 'local'} Parsoid`);
+      if (!this.mwCapabilities.veApiAvailable) {
+        logger.log(`Using local MCS and ${this.mwCapabilities.veApiAvailable} local Parsoid`);
```

Comment:
This log entry should not be there but in `setBaseUrls()` which is the part which decides which API (mobilerest, desktoprest, ve, parsoid,... ) we are using.